provisioning/device/src/tpm_registration.ts

```diff
@@ -0,0 +1,239 @@
+import { EventEmitter } from 'events';
+import * as machina from 'machina';
+import * as dbg from 'debug';
+const debug = dbg('azure-iot-provisioning-device:TpmRegistration');
+import { RegistrationClient, TpmProvisioningTransport, TpmSecurityClient, TpmRegistrationInfo } from './interfaces';
+
+export class TpmRegistration extends EventEmitter implements RegistrationClient {
+  private _fsm: machina.Fsm;
+  private _transport: TpmProvisioningTransport;
+  private _securityClient: TpmSecurityClient;
+  private _idScope: string;
+
+  constructor(idScope: string, transport: TpmProvisioningTransport, securityClient: TpmSecurityClient) {
```

Comment:
>idScope: string [](start = 14, length = 15)

For X509, I made the scope a parameter of the transport constructor.  I did this because the transport is the only component that needs this.  e.g. the X509Registration object doesn't need to know about it.  We should synchronize here.  I could certainly move it to the client constructor.

Reply:
that's a good question... The only place we use the idScope is to generate the SAS. I'm not sure yet if the format of the SAS is generic or transport-specific. Let's see how that pans out with an actual transport implementation.

---
In reply to: [151768228](https://github.com/Azure/azure-iot-sdk-node/pull/155#discussion_r151768228) [](ancestors = 151768228)